cmd/curator/curator.go

```diff
@@ -150,30 +157,85 @@ func curatorRun(config *rest.Config, client *clientv1.Client, clusterName string
 
 	}
 
-	if strings.Contains(jobChoice, "activate-and-monitor") {
+	if jobChoice == "activate-and-monitor" {
 		hiveset, err := hiveclient.NewForConfig(config)
 		utils.CheckError(err)
 
-		err = hive.ActivateDeploy(hiveset, clusterName)
-		utils.CheckError(err)
+		if err = hive.ActivateDeploy(hiveset, clusterName); err != nil {
+			utils.CheckError(utils.RecordFailedCuratorStatusCondition(
+				*client,
+				clusterName,
+				jobChoice,
+				v1.ConditionTrue,
+				err.Error()))
+			klog.Error(err.Error())
+			panic(err)
+		}
 	}
 
-	if strings.Contains(jobChoice, "monitor") {
```

Comment:
@leena-jawale I'm pretty sure this was a cheat I was using to run this code for `monitor` and `activate-and-monitor`.  So you probably need to leave this one as "contains", or at the end of activate-and-monitor, set the jobChoice = "monitor"